File: JUNE-2025/30-06-2025/p594.rb
def find_lhs(nums)
  freq = Hash.new(0)
  nums.each { |n| freq[n] += 1 }

  ans = 0
  freq.each do |key, val|
    if freq.key?(key + 1)
      ans = [ans, val + freq[key + 1]].max
    end
  end

  ans
end

File: JUNE-2025/30-06-2025/p594.rs
use std::collections::HashMap;
use std::cmp::max;

impl Solution {
    pub fn find_lhs(nums: Vec<i32>) -> i32 {
        let mut freq = HashMap::new();
        for &num in &nums {
            *freq.entry(num).or_insert(0) += 1;
        }

        let mut ans = 0;
        for &key in freq.keys() {
            if let Some(&val) = freq.get(&(key + 1)) {
                ans = max(ans, freq[&key] + val);
            }
        }

        ans
    }
}